### PR TITLE
add deploy yamls for dm with new ha architecture

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 
add deploy yamls for dm with new ha architecture
### What is changed and how does it work?
add deploy yamls for dm with new ha architecture
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: manifests/dm/master/dm-master.yaml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      - ReadWriteOnce
+      dataSource: null
+      resources:
+        requests:
+          storage: 10Gi
+      storageClassName: local-storage
+    status:
+      phase: Pending
+---
+    apiVersion: v1
```
Comment (tennix) [inline manifests/dm/master/dm-master.yaml]: Why indent here?

File: manifests/dm/master/dm-master.yaml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      dataSource: null
+      resources:
+        requests:
+          storage: 10Gi
+      storageClassName: local-storage
+    status:
+      phase: Pending
```
Comment (tennix) [inline manifests/dm/master/dm-master.yaml]: This field is unnecessary.

File: manifests/dm/master/dm-master.yaml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      schedulerName: default-scheduler
+      securityContext: {}
+      terminationGracePeriodSeconds: 30
+  updateStrategy:
+    rollingUpdate:
+      partition: 3
+    type: RollingUpdate
+  volumeClaimTemplates:
+  - metadata:
+      creationTimestamp: null
```
Comment (tennix) [inline manifests/dm/master/dm-master.yaml]: Unnecessary field.

File: manifests/dm/master/dm-master.yaml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: PEER_SERVICE_NAME
+          value: dm-master-peer
+        - name: SERVICE_NAME
+          value: dm-master
+        - name: TZ
+          value: UTC
+        image: 127.0.0.1:30002/pingcap/dm:2040
```
Comment (tennix) [inline manifests/dm/master/dm-master.yaml]: There is already a DM ha-alpha version on DockerHub.

**Comment by tennix** [comment]
I think we need to provide a simple example configmap for DM.

File: manifests/dm/master/dm-master.yaml (hunk 1 above)
Comment (DanielZhangQD) [reply manifests/dm/master/dm-master.yaml]: fixed.

File: manifests/dm/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+---
```
Comment (cofyc) [inline manifests/dm/README.md]: move docs to `docs/` directory?

File: manifests/dm/master/dm-master.yaml (hunk 3 above)
Comment (DanielZhangQD) [reply manifests/dm/master/dm-master.yaml]: Removed.

File: manifests/dm/master/dm-master.yaml (hunk 4 above)
Comment (DanielZhangQD) [reply manifests/dm/master/dm-master.yaml]: Updated.

File: manifests/dm/README.md (hunk 5 above)
Comment (DanielZhangQD) [reply manifests/dm/README.md]: This is just for temp deployment, we may need to support DM with chart or controller later, I think we can leave it here for now.

**Comment by DanielZhangQD** [comment]
> I think we need to provide a simple example configmap for DM.

With new HA architecture, 

- For dm-master, there are only few [RPC](https://github.com/pingcap/dm/blob/ha-dev/dm/master/dm-master.toml) parameters that can be configured, and I have consulted with @csuzhangxc , no need to expose them to users, they have not met any users that need to customize them till now, all the other configurations can be done with command-line options and dmctl.

- For dm-worker, all the [configs](https://github.com/pingcap/dm/blob/ha-dev/dm/worker/dm-worker.toml) can be done with command-line options and no need to expose to users.

File: manifests/dm/master/dm-master.yaml (hunk 2 above)
Comment (DanielZhangQD) [reply manifests/dm/master/dm-master.yaml]: Removed

**Comment by DanielZhangQD** [comment]
> I think we need add configmap to configure the dm-worker and dm-master.

Please check the comments [here](https://github.com/pingcap/tidb-operator/pull/1738#issuecomment-589519307).

**Comment by DanielZhangQD** [comment]
@tennix @Yisaer configmap for dm-master is added, no need for dm-worker, PTAL again.

**Pull request merged**